lib/internal/http2/core.js

```diff
@@ -1528,15 +1521,12 @@ function continueStreamDestroy(self, err, callback) {
   // Unenroll the timer
   self.setTimeout(0);
 
-  setImmediate(finishStreamDestroy, self, handle);
+  setImmediate(finishStreamDestroy, self, session[kHandle]);
 
-  // All done
-  const rst = state.rst;
-  const code = rst ? state.rstCode : NGHTTP2_NO_ERROR;
   // RST code 8 not emitted as an error as its used by clients to signify
   // abort and is already covered by aborted event, also allows more
   // seamless compatibility with http1
-  if (!err && code !== NGHTTP2_NO_ERROR && code !== NGHTTP2_CANCEL) {
+  if (code !== NGHTTP2_NO_ERROR && code !== NGHTTP2_CANCEL && !err) {
```

Comment:
Assume that `rst` is falsy and `session.destroyed` is truthy (not sure if this can happen). In this case `code === state.rstCode`. Before the change we had `code === NGHTTP2_NO_ERROR`.

Reply:
Yep, that's correct. `state.rstCode` would be NGHTTP2_NO_ERROR in that situation.

Reply:
Oh ok, ignore me then.